### PR TITLE
feat(m365): add CIS M365 v7.0.0 exchange checks

### PR DESCRIPTION
**Pull request by pedrooot**
### Context

Part of the split of #12102, which added 37 M365 checks in a single pull request. It was broken up per service so each piece can be reviewed on its own; this one covers **exchange**.

The CIS Microsoft 365 Foundations Benchmark v7.0.0 framework (`cis_7.0_m365`) defines 160 controls, and 64 of its "Automated" controls had no Prowler check.

### Description

Adds **2 M365 `exchange` checks** mapped to `cis_7.0_m365.json`:

| CIS control | Check |
|---|---|
| 6.3.2 | `exchange_owa_mailbox_policy_personal_accounts_disabled` |
| 6.5.5 | `exchange_organization_reject_direct_send_enabled` |

**Data collection.** No new getters: only new fields on the existing `Organization` and `MailboxPolicy` models (`RejectDirectSend`, `PersonalAccountsEnabled`, `IsDefault`).

Each check ships with its implementation, `metadata.json` and unit tests (PASS / FAIL / edge cases).

### Steps to review

1. **Discovery:** `uv run python prowler-cli.py m365 --list-checks | grep -E "exchange_owa_mailbox_policy_personal_accounts_disabled|exchange_organization_reject_direct_send_enabled"`
2. **Tests:** `uv run pytest tests/providers/m365/services/exchange/`
3. **Compliance mapping:** review the 2 requirement(s) touched in `prowler/compliance/m365/cis_7.0_m365.json`.
5. Spot-check the check logic against the CIS audit procedures (fail-closed defaults; Conditional Access report-only treated as FAIL).

### Checklist

<details>

<summary><b>Community Checklist</b></summary>

- [ ] This feature/issue is listed in the [open issues](https://github.com/prowler-cloud/prowler/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen) or roadmap.prowler.com
- [ ] Is it assigned to me, if not, request it via the [open issues](https://github.com/prowler-cloud/prowler/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen) or [Prowler Community Slack](https://goto.prowler.com/slack)

</details>

- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.
- [ ] Review if is needed to change the [README.md](https://github.com/prowler-cloud/prowler/blob/master/README.md)
- [x] Ensure a changelog fragment is added under [prowler/changelog.d/](https://github.com/prowler-cloud/prowler/tree/master/prowler/changelog.d), if applicable.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

#### SDK/CLI
- Are there new checks included in this PR? **Yes** (2 M365 checks)
    - No new permissions: reuses the Exchange Online / Teams sessions the provider already establishes.

## Summary by CodeRabbit

- **New Features**
  - Added Microsoft 365 Exchange security checks aligned with CIS Microsoft 365 Foundations Benchmark v7.0.0.
  - Detects Exchange Online configurations that allow Direct Send.
  - Detects personal accounts or calendars enabled in the default Outlook on the web mailbox policy.
  - Includes remediation guidance and risk details for each check.

- **Bug Fixes**
  - Suppressed misleading errors when no application access policies exist.

- **Tests**
  - Added coverage for compliant, non-compliant, missing, and ignored Exchange configurations.

### CODE REVIEW

File: prowler/providers/m365/lib/powershell/m365_powershell.py [hunk 1]
```diff
@@ -1022,8 +1022,11 @@ def get_application_access_policies(self) -> dict:
                 }
             ]
         """
+        # -ErrorAction SilentlyContinue: tenants with no application access
+        # policies raise a localized "object not found" error instead of
+        # returning an empty result; the error output never carries data.
         return self.execute(
-            "Get-ApplicationAccessPolicy | ConvertTo-Json -Depth 10",
+            "Get-ApplicationAccessPolicy -ErrorAction SilentlyContinue | ConvertTo-Json -Depth 10",
```
[the remark this answers is not in the document]
Comment (danibarranqueroo) [reply prowler/providers/m365/lib/powershell/m365_powershell.py:1029]: Not applying this one — the premise doesn't hold:

1. `-ErrorAction SilentlyContinue` did not change behavior: PowerShell stderr never fed the `None`/MANUAL path (that only triggers on Python exceptions in `_get_application_access_policies`), before or after this change. Only the log line was suppressed.
2. The object-not-found error **is** the normal response for a tenant with zero application access policies — Exchange Online errors instead of returning an empty result. In that case evaluating with an empty list is the correct outcome, not MANUAL.
3. Matching the error text is not robust: the message is localized per tenant language (the one that motivated this fix arrived in Spanish).

The empty-result vs. transport-failure ambiguity predates this PR and would need structured error output from EXO to resolve reliably.

**Review by danibarranqueroo — APPROVED** [review]
